featuretools/primitives/standard/datetime_transform_primitives.py

```diff
@@ -331,6 +334,72 @@ def is_leap_year(vals):
         return is_leap_year
 
 
+class IsLunchTime(TransformPrimitive):
+    """Determines if a datetime falls during lunch time (non-holiday weekday with hour=12)
+
+    Args:
+        country (str): Country to use for determining Holidays.
+            Default is 'US'. Should be one of the available countries here:
+            https://github.com/dr-prodigy/python-holidays#available-countries
+        include_weekend (bool): Include weekend dates as datetimes with valid
+            lunch times. Default is True.
+        include_holidays (bool): Include holiday dates as datetimes with valid
+            lunch times. Default is False.
+
+    Examples:
+        >>> from datetime import datetime
+        >>> dates = [datetime(2022, 6, 21, 12, 3, 3),
+        ...          datetime(2019, 1, 3, 4, 4, 4),
+        ...          datetime(2022, 1, 1, 12, 1, 2),
+        ...          np.nan]
+        >>> ilt = IsLunchTime()
+        >>> ilt(dates).tolist()
+        [True, False, False, False]
+    """
+
+    name = "is_lunch_time"
+    input_types = [ColumnSchema(logical_type=Datetime)]
+    return_type = ColumnSchema(logical_type=BooleanNullable)
+    compatibility = [Library.PANDAS, Library.DASK, Library.SPARK]
+    description_template = "whether {} falls during lunch time"
+
+    def __init__(self, country="US", include_weekends=True, include_holidays=False):
+        self.country = country
+        years_list = [1950 + x for x in range(150)]
+        self.federal_holidays = getattr(holidays, self.country)(years=years_list)
+        self.include_weekends = include_weekends
+        self.include_holidays = include_holidays
+        self.holidays_df = pd.DataFrame(
+            sorted(self.federal_holidays.items()), columns=["dates", "names"]
+        )
+        self.timestamp_dates = [pd.Timestamp(i) for i in self.holidays_df.dates]
```

Comment:
If we have `include_holidays` default to `False` we only need to do the holiday related setup in this block when a user creates an instance with `include_holidays=True`. Otherwise we are doing unnecessary work here that never gets used later.

Reply:
Also `self.timestamp_dates` only seems to be used in the `spark_mask` function, so there we don't need to calculate it here and store it on the primitive object - we can calculate that inside `spark_mask` instead.